### PR TITLE
Cleanup, remove abseil dependency, compile on windows with Ninja

### PR DESCRIPTION
**Pull request by ugermann**
This pull request tidies up some loose ends

- The dependency on abseil for string_view with obsolete compilers is eliminated.
- Cleaner cmake setup.
- Compiles on windows with ninja generator.
- Probably compiles on Gentoo/Arch Linux out of the box with dynamic prce2 library (not tested)

Setups tested (see workflows files for most cases, except the valhalla setup):
- Compiles on valhalla (Ubuntu 16.04 with cmake 3.5 and gcc 5.5 / clang 3.8).
- Compiles on windows with the Ninja generator.
- Compiles on the mac github runner.
- Compiles on Ubuntu 20.04 with clang, g++, and emscripten for WASM compiles.

### CODE REVIEW

File: CMakeLists.txt
Comment (XapaJIaMnu) [inline CMakeLists.txt:14]: Why do we sometimes use C++17 and sometimes C++14? We can manually set C++17, by appending `-std=c++17` to CXXFLAGS (and some ifdefs of MSVC) if necessary. My understanding is that if we have C++17, the work with maps would be slightly easier for other people? We have C++17 with everything but the clang on 16.04. Would it make sense to enforce c++17 , or is someone (emscripten) preventing us from doing it?

**Review by XapaJIaMnu — APPROVED** [review]
Other than not quite understanding why we use CXX_SET_STANDARD as opposed to explicitly setting it, I think it's good.

File: CMakeLists.txt
Comment (ugermann) [reply CMakeLists.txt:14]: Simplicity, of sorts. In my opinion, there should be a single mechanism to enforce the c++ standard, rather than setting cxx flags explicitly. Using CMAKE_CXX_STANDARD dumps the responsibility of setting flags according to OS / Compiler onto cmake. This is a fringe case anyway that will hopefully go away soon once we upgrade valhalla to an actually supported versoin of Ubuntu. On a marginal note: I've jumped the gun and issued a pull request on browsermt/ssplit-cpp; with a few minor fixes, it's the same as this PR. Jerin has approved the PR; once you approve, I'll merge. Thanks, by the way, for all the work you've put into making this compile on Windows and overall cleanup.

**Pull request merged**
